### PR TITLE
Custom CSS: Removing Custom CSS Legacy Migration

### PR DESCRIPTION
**Pull request by coder-karen**
#### Changes proposed in this Pull Request:

* In WordPress 4.7, WordPress Core added Additional CSS which meant that we then extended their solution. For sites running earlier versions, we still used our custom solution, so had a migration in place.
* As it has been more than 6 years since WordPress 4.7, we're now removing that migration so that the feature only extends Core's solution (and otherwise won't work).

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

https://github.com/Automattic/jetpack/pull/27084#discussion_r1007940891

p9dueE-4t8-p2 

#### Does this pull request change what data or activity we track or use?

No.

#### Questions

* In this PR we're both removing and renaming files (`custom-css-4.7.php` is renamed to `custom-css.php`, and the `custom-css.php` in it's previous place is removed). Will this cause any issues when doing the weekly WordPress.com release - would it be wiser to do the renaming separately (after the first PR is merged and released)?

#### Testing instructions:

* Test the Custom CSS feature. In particular test with custom CSS already added to make sure nothing (and no settings) are lost while upgrading to this feature branch.

### CODE REVIEW

**Review by kraftbj — DISMISSED** [review]

**Comment by anomiex** [comment]
> Will this cause any issues when doing the weekly WordPress.com release - would it be wiser to do the renaming separately (after the first PR is merged and released)?

No, but we should be sure to replace the stub at `wp-content/mu-plugins/custom-css.php` on wpcom with the contents of the deleted file at some point before the weekly release. Probably do it as soon as this is merged.

Looks like we could also delete a few other files:
* projects/plugins/jetpack/modules/custom-css/custom-css/css/css-editor.css
* projects/plugins/jetpack/modules/custom-css/custom-css/css/rtl/codemirror-rtl.css
* projects/plugins/jetpack/modules/custom-css/custom-css/css/use-codemirror.css
* projects/plugins/jetpack/modules/custom-css/custom-css/js/core-customizer-css.js
* projects/plugins/jetpack/modules/custom-css/custom-css/js/css-editor.js
* projects/plugins/jetpack/modules/custom-css/custom-css/js/use-codemirror.js

We'll probably also need to restore some of those to `wp-content/mu-plugins/custom-css/js` and `wp-content/mu-plugins/custom-css/css`.

**Comment by coder-karen** [comment]
> No, but we should be sure to replace the stub at wp-content/mu-plugins/custom-css.php on wpcom with the contents of the deleted file at some point before the weekly release. 

Not sure I fully understand here. Which deleted file would we be replacing the contents of `wp-content/mu-plugins/custom-css.php` with?

> Looks like we could also delete a few other files:

Ah yes! Looks like I'd need to follow up with deleting all but `css/rtl/codemirror-rtl.css` and `js/core-customizer-css.js` later as in testing the build won't complete otherwise.

**Comment by anomiex** [comment]
> Not sure I fully understand here. Which deleted file would we be replacing the contents of `wp-content/mu-plugins/custom-css.php` with?

[The former `projects/plugins/jetpack/modules/custom-css/custom-css.php`](https://github.com/Automattic/jetpack/blob/e939eb1d92589ebfc1592430b25b143dd599b485/projects/plugins/jetpack/modules/custom-css/custom-css.php), which is being deleted in this PR and replaced with `custom-css-4.7.php`.

> Ah yes! Looks like I'd need to follow up with deleting all but `css/rtl/codemirror-rtl.css` and `js/core-customizer-css.js` later as in testing the build won't complete otherwise.

Ah, that's because of
https://github.com/Automattic/jetpack/blob/012f5bdc3b4fcfbf02e3ed74d5c288c5b1299d99/projects/plugins/jetpack/tools/webpack.config.css.js#L172-L173
Removing those lines should fix the build.

File: projects/plugins/jetpack/modules/custom-css/custom-css.php
Comment (anomiex) [inline projects/plugins/jetpack/modules/custom-css/custom-css.php:46]: Looks like the rebase managed to lose one closing brace in the file
```suggestion
		add_action( 'admin_init', array( __CLASS__, 'set_content_width' ) );
	}
```

**Comment by coder-karen** [comment]
> Removing those lines should fix the build.

Excellent, thank you.

On the Jetpack side, everything works well in testing with the extra removed files / changes.

> The former projects/plugins/jetpack/modules/custom-css/custom-css.php, which is being deleted in this PR and replaced with custom-css-4.7.php.

I'm still not following what's needed on the WPcom side - adding back the contents of that file I notice there is a trickle down effect. It results in needing to add back preprocessors.php (though I can make it a stub), then there is an issue with CSS Tidy so it looks like this diff needs deploying first: D70703-code. That diff will add back the contents of custom-css.php too. I'd be happy to jump on a quick call sometime next week too just to wrap my head around all of this :)

**Comment by anomiex** [comment]
D70703-code is just out of date, it won't merge as-is. Right now if they rebase it they'll wind up removing their change to `custom-css.php` entirely. If we make the change needed here first, when they rebase it should just work.

What's needed on the wpcom side should be something like this:
* `git checkout f0a2dac982888129fbe70691d28e4f5f0f0f5c9c^ wp-content/mu-plugins/custom-css.php` to restore the prior version of that file.
  * Alternatively, `cp wp-content/mu-plugins/jetpack-plugin/production/modules/custom-css/custom-css.php wp-content/mu-plugins/custom-css.php` then fix some things up that need fixing, which will likely wind up being about the same thing.
* Then restore several of the files being deleted here that were also deleted in D90585-code, or copy the identical versions back from the current Jetpack.
  * wp-content/mu-plugins/custom-css/css/css-editor.css
  * wp-content/mu-plugins/custom-css/css/use-codemirror.css
  * wp-content/mu-plugins/custom-css/js/css-editor.js
  * wp-content/mu-plugins/custom-css/js/use-codemirror.js
* Fix up some more things in wp-content/mu-plugins/custom-css.php to refer to Jetpack copies of the file instead, just like we did for stuff when de-Fusioning. (the line numbers shown here may be slightly off from the wpcom copy, but the lines are all the same)
  * https://github.com/Automattic/jetpack/blob/b5883c969166b708a31ce5ad11067d147330a558/projects/plugins/jetpack/modules/custom-css/custom-css.php#L228
  * Possibly https://github.com/Automattic/jetpack/blob/b5883c969166b708a31ce5ad11067d147330a558/projects/plugins/jetpack/modules/custom-css/custom-css.php#L924
  * https://github.com/Automattic/jetpack/blob/b5883c969166b708a31ce5ad11067d147330a558/projects/plugins/jetpack/modules/custom-css/custom-css.php#L1090
  * https://github.com/Automattic/jetpack/blob/b5883c969166b708a31ce5ad11067d147330a558/projects/plugins/jetpack/modules/custom-css/custom-css.php#L1093
  * https://github.com/Automattic/jetpack/blob/b5883c969166b708a31ce5ad11067d147330a558/projects/plugins/jetpack/modules/custom-css/custom-css.php#L1893
  * https://github.com/Automattic/jetpack/blob/b5883c969166b708a31ce5ad11067d147330a558/projects/plugins/jetpack/modules/custom-css/custom-css.php#L1954

**Comment by coder-karen** [comment]
Gotcha, thank you! D92421-code should do it (confirmed new Jetpack file paths work) - could you please double check just to make sure? 🙇‍♀️  Once that's got an approval I'll make sure this PR is ready for review again, then get this merged followed by deploying the diff.

**Review by anomiex — APPROVED** [review]

**Pull request merged**
